### DISCUSSION TITLE
avif support within DZI

### DISCUSSION THREAD

**Discussion by stecca21** [openseadragon/openseadragon#2439] [General]
Is there any chance of supporting avif format under DZI?

**Comment by msalsbery** [comment]
Yes!

**Comment by msalsbery** [comment]
;)

I don't know how we'd go about changing the actual DZI spec, but browser support is good, and the changes to OpenSeadragon are trivial.

**Comment by stecca21** [comment]
actually, it looks like adding the extension works just fine, at least on initial testing:
`OpenSeadragon.setImageFormatsSupported({avif: true});`

**Reply by msalsbery** [reply]
> actually, it looks like adding the extension works just fine, at least on initial testing:
> 
> `OpenSeadragon.setImageFormatsSupported({avif: true});`

That IS trivial!

Thanks for the update!

**Reply by iangilman** [reply]
Cool! I figure we can add this to the official list once Edge supports AVIF. Meanwhile, it's easy to add per project :)

**Reply by msalsbery** [reply]
> Cool! I figure we can add this to the official list once Edge supports AVIF. Meanwhile, it's easy to add per project :)

Microsoft...Making My Life Miserable Since DOS 3.0

**Reply by stecca21** [reply]
I thought that already happened...sort of:
https://www.msftnext.com/how-to-enable-avif-format-support-in-microsoft-edge/

**Reply by msalsbery** [reply]
> I thought that already happened...sort of:
> 
> https://www.msftnext.com/how-to-enable-avif-format-support-in-microsoft-edge/
> 
> 

The most I could find is anecdotal evidence it might work but only on Windows 11. I don't have any Windows 11 machines so i don't know until I finally upgrade hardware some day

Edit: no love with webkit on my iPhone 7 either

**Reply by stecca21** [reply]
The iPhone 7 won't see support as it was introduced in the first iOS that dropped support for the phone.

I tried to see if Edge for mac made that startup flag available through the front end flag mgmt page and did not see it. I'm currently hunting around for the proper place to manually include the flag.

**Reply by iangilman** [reply]
Good to know it's possible! We won't add it officially until it's on by default in Edge, of course.

@msalsbery Bravo on not jumping on the latest iPhone every time they release a new one! I'm pleased with how well the old ones are holding up (AVIF notwithstanding).

**Reply by stecca21** [reply]
Getting closer: https://www.neowin.net/news/microsoft-edge-121-finally-brings-avif-support/

**Reply by iangilman** [reply]
Cool! Let's keep an eye on it :)

**Reply by bmccarthyco** [reply]
Now that Edge support is released, any updates on what's next for official support?

**Reply by iangilman** [reply]
@bmccarthynn Thanks for keeping an eye on it! We should go ahead and add it. It's just a matter of adding it here: 

https://github.com/openseadragon/openseadragon/blob/9b7fc464ca3ae3fdd9f981b64aea5d8348021616/src/openseadragon.js#L2700

You're welcome to make a PR if you'd like :)

**Reply by msalsbery** [reply]
PR submitted #2544

**Reply by msalsbery** [reply]
@stecca21 

AVIF and WebP added to supported formats in 5.0.0 (in case you missed it)!
